### PR TITLE
feat(widget): add widget component

### PR DESCRIPTION
**Pull request by brian-patrick-3**

### CODE REVIEW

**Comment by maulik-thaker1** [comment]
Curious about this `disabled` functionality. Here only background is lighten but we can still able to click widget internals when `disabled = true`. Is it as needed?

Uploading Screen Recording 2024-04-26 at 5.01.59 PM.mov…

File: src/components/reusable/widget/widget.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  @property({ type: String })
+  widgetTitle = '';
+
+  /** Widget sub-title. */
+  @property({ type: String })
+  subTitle = '';
+
+  /** Widget drag active state. */
+  @property({ type: Boolean })
+  dragActive = false;
```
Comment (maulik-thaker1) [inline src/components/reusable/widget/widget.ts:27]: What else `dragActive` do apart from adding shadow ?

Comment (brian-patrick-3) [reply src/components/reusable/widget/widget.ts:27]: That's it.

**Comment by brian-patrick-3** [comment]
> Curious about this `disabled` functionality. Here only background is lighten but we can still able to click widget internals when `disabled = true`. Is it as needed?

Yeah makes sense because we don't have control over the slotted content.

**Pull request merged**